src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,578 @@
+    bytes32 lastSeenHash;
+  }
+
+  struct StrikeData {
+    uint strike;
+    uint expiry;
+    int markPrice;
+    uint tradePrice;
+    int tradeAmount;
+  }
+
+  // keccak256(abi.encode(uint256(keccak256("lyra.storage.PrincipalProtectedTSA")) - 1)) & ~bytes32(uint256(0xff))
+  bytes32 private constant PPTSAStorageLocation = 0x273fadd8b9727c49bd8eabddca41871e1e7fc468f40ece57b00a1bd25002e500;
+
+  function _getPPTSAStorage() private pure returns (PPTSAStorage storage $) {
+    assembly {
+      $.slot := PPTSAStorageLocation
+    }
+  }
+
+  constructor() {
+    _disableInitializers();
+  }
+
+  function initialize(
+    address initialOwner,
+    BaseTSA.BaseTSAInitParams memory initParams,
+    PPTSAInitParams memory ppInitParams
+  ) external reinitializer(2) {
+    __BaseTSA_init(initialOwner, initParams);
+
+    PPTSAStorage storage $ = _getPPTSAStorage();
+
+    $.depositModule = ppInitParams.depositModule;
+    $.withdrawalModule = ppInitParams.withdrawalModule;
+    $.rfqModule = ppInitParams.rfqModule;
+    $.tradeModule = ppInitParams.tradeModule;
+    $.optionAsset = ppInitParams.optionAsset;
+    $.baseFeed = ppInitParams.baseFeed;
+    $.isCallSpread = ppInitParams.isCallSpread;
+    $.isLongSpread = ppInitParams.isLongSpread;
+    BaseTSAAddresses memory tsaAddresses = getBaseTSAAddresses();
+
+    tsaAddresses.depositAsset.approve(address($.depositModule), type(uint).max);
+  }
+
+  ///////////
+  // Admin //
+  ///////////
+  function setPPTSAParams(CollateralManagementParams memory newCollateralMgmtParams, PPTSAParams memory pptsaParams)
+    external
+    onlyOwner
+  {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    if (
+      pptsaParams.minSignatureExpiry < 1 minutes || pptsaParams.minSignatureExpiry > pptsaParams.maxSignatureExpiry
+        || newCollateralMgmtParams.worstSpotBuyPrice < 1e18 || newCollateralMgmtParams.worstSpotBuyPrice > 1.2e18
+        || newCollateralMgmtParams.worstSpotSellPrice > 1e18 || newCollateralMgmtParams.worstSpotSellPrice < 0.8e18
+        || newCollateralMgmtParams.spotTransactionLeniency < 1e18
+        || newCollateralMgmtParams.spotTransactionLeniency > 1.2e18
+        || pptsaParams.minMarkValueToStrikeDiffRatio > pptsaParams.maxMarkValueToStrikeDiffRatio
+        || pptsaParams.maxMarkValueToStrikeDiffRatio > 1e18 || pptsaParams.maxMarkValueToStrikeDiffRatio < 1e16
+        || pptsaParams.strikeDiff > 2e21 || pptsaParams.strikeDiff < 1e20
+        || pptsaParams.optionMaxTimeToExpiry <= pptsaParams.optionMinTimeToExpiry
+        || newCollateralMgmtParams.feeFactor > 0.05e18 || pptsaParams.maxTotalCostTolerance < 2e17
+        || pptsaParams.maxTotalCostTolerance > 5e18 || pptsaParams.maxLossPercentOfTVL < 1e14
+        || pptsaParams.maxLossPercentOfTVL > 1e18 || pptsaParams.negMaxCashTolerance < 1e16
+        || pptsaParams.negMaxCashTolerance > 1e18 || ($.isLongSpread && pptsaParams.maxTotalCostTolerance < 1e18)
```

Comment:
would double check these dual boolean checks too and make sure they make sense. Vlad might say forcing long call spread vaults to have `maxTotalCostTolerance > 1e18` might be actually too aggressive.

Reply:
+1 definitely better to make these looser than needed. They're primarily there to prevent fatfingers/completely invalid values. Anything that might be bad from a financial standpoint should be possible to be set

Reply:
i.e. strike diff probably doesn't even need to be checked. You're hardcoding it to a pretty specific use case of >100 & <2k wide spreads. That won't necessarily work for bitcoin (on the upper end) or solana options (on the lower end)

Reply:
Verified in the group chat that these bounds are alrighty.